src/Connections/GitlabRepo.ts

```diff
@@ -222,13 +222,17 @@ export class GitLabRepoConnection extends CommandConnection {
             return;
         }
         const branchname = event.ref.replace("refs/heads/", "");
-        const commitsurl = `${event.project.homepage}/-/commit/${event.after}`
+        const commitsurl = `${event.project.homepage}/-/commits/${branchname}`;
         const branchurl = `${event.project.homepage}/-/tree/${branchname}`;
         const shouldName = !event.commits.every(c => c.author.name === event.commits[0]?.author.name);
```

Comment:
@Half-Shot: I think this is wrong, it should rather check if the pusher and commiter are the same.
The way it currently is, I can pull a commit from upstream and push it to my forked remote and it will only tell me that I pushed it but not the original commiter, who is someone else (actually happened today).

Reply:
Sure, I think that's reasonable. Perhaps just change the if statement to check against the pusher email address? (since that's the best unique thing we have)

Reply:
so, technically shouldName could also depend on tooManyCommits, but i'm not sure it's worth doing